src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -67,25 +60,10 @@ internal sealed class RestoreOperationLogger : LegacyLoggerAdapter, ILogger, IDi
             JoinableTaskFactory jtf,
             CancellationTokenSource cts)
         {
-            if (errorListDataSource == null)
-            {
-                throw new ArgumentNullException(nameof(errorListDataSource));
-            }
-
-            if (jtf == null)
-            {
-                throw new ArgumentNullException(nameof(jtf));
-            }
-
-            if (cts == null)
-            {
-                throw new ArgumentNullException(nameof(cts));
-            }
-
             _operationSource = operationSource;
-            _errorListDataSource = errorListDataSource;
-            _taskFactory = jtf;
-            _externalCts = cts;
+            _errorListDataSource = errorListDataSource ?? throw new ArgumentNullException(nameof(errorListDataSource));
+            _taskFactory = jtf ?? throw new ArgumentNullException(nameof(jtf));
+            _externalCts = cts ?? throw new ArgumentNullException(nameof(cts));
```

Comment:
We started using `Assumes` API everywhere else 'cause it's less ugly than this nonsense :)